backend/src/main/java/de/neuefische/henny/reisekasse/service/EventService.java

```diff
@@ -182,4 +180,76 @@ public Event getEventById(String eventId) {
         return eventMembers;
     }
 
-}
\ No newline at end of file
+    public List<Transfer> compensateBalances(List<EventMember> eventMembers) {
+
+        List<EventMember> paymentReceivers = eventMembers.stream()
+                .filter(member -> member.getBalance() > 0)
+                .sorted(Comparator.comparing(EventMember::getBalance))
+                .collect(Collectors.toList());
+
+        List<EventMember> payers = eventMembers.stream()
+                .filter(member -> member.getBalance() < 0)
+                .sorted(Comparator.comparing(EventMember::getBalance))
+                .collect(Collectors.toList());
+        Collections.reverse(payers);
```

Comment:
You can make use of the reversed method [java docs](https://docs.oracle.com/javase/8/docs/api/java/util/Comparator.html#reversed--)
```suggestion
                .sorted(Comparator.comparing(EventMember::getBalance).reversed())
                .collect(Collectors.toList());
```